templates/web-application/webpack.dev.js

```diff
@@ -0,0 +1,89 @@
+// @ts-check
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import HtmlWebpackPlugin from "html-webpack-plugin";
+import ReactRefreshWebpackPlugin from "@pmmmwh/react-refresh-webpack-plugin";
+import open from "open";
+
+import { loadSwcConfig } from "./loadSwcConfig.js";
+
+/** @type {import("webpack").Configuration} */
+
+class OpenBrowser {
+    isFirstCompile = true;
+
+    apply(compiler) {
+        compiler.hooks.done.tap("done", async () => {
+            if (this.isFirstCompile) {
+                console.log("\nOpening browser at: \"http://localhost:8080/\"");
+
+                await open("http://localhost:8080/");
+            }
+            this.isFirstCompile = false;
+        });
+    }
+}
+
+export default {
+    mode: "development",
+    target: "web",
+    devtool: "eval-cheap-module-source-map",
+    devServer: {
+        port: 8080,
+        historyApiFallback: true,
+        hot: true
+    },
+    entry: "./src/index.tsx",
+    output: {
+        // The trailing / is very important, otherwise paths will not be resolved correctly.
+        publicPath: "http://localhost:8080/"
+    },
+    cache: {
+        type: "filesystem",
+        allowCollectingMemory: true,
+        buildDependencies: {
+            config: [fileURLToPath(import.meta.url)]
+        },
+        cacheDirectory: path.resolve("node_modules/.cache/webpack")
+    },
+    module: {
+        rules: [
+            {
+                test: /\.(js|ts|tsx)$/i,
+                exclude: /node_modules/,
+                include: path.resolve("src"),
```

Comment:
I add the include here too, as specified above.

Reason: https://webpack.js.org/guides/build-performance/#loaders

Reply:
Might be worth to actually add those comments in the config so future maintainers understand why it's configured this way.